cdap-watchdog/src/main/java/co/cask/cdap/logging/framework/LogPipelineLoader.java

```diff
@@ -0,0 +1,234 @@
+                             "The CDAP system log processing pipeline is missing. " +
+                               "Please check and fix any configuration error shown earlier in the log.");
+    return result;
+  }
+
+  /**
+   * Returns an {@link Iterable} of {@link URL} of pipeline configuration files.
+   */
+  private Iterable<URL> getPipelineConfigURLs() {
+    URL systemPipeline = getClass().getClassLoader().getResource(CDAP_LOG_PIPELINE_CONFIG);
+    // This shouldn't happen since the cdap pipeline is packaged in the jar.
+    Preconditions.checkState(systemPipeline != null, "Missing cdap system pipeline configuration");
+
+    List<File> files = DirUtils.listFiles(new File(cConf.get(Constants.Logging.PIPELINE_CONFIG_DIR)), "xml");
+    return Iterables.concat(Collections.singleton(systemPipeline),
+                            Iterables.filter(Iterables.transform(files, new Function<File, URL>() {
+      @Nullable
+      @Override
+      public URL apply(File file) {
+        try {
+          return file.toURI().toURL();
+        } catch (MalformedURLException e) {
+          // This shouldn't happen
+          LOG.warn("Ignore log pipeline config file {} due to {}", file, e.getMessage());
+          return null;
+        }
+      }
+    }), Predicates.notNull()));
+  }
+
+  /**
+   * Creates the {@link LogPipelineSpecification} representing the given log pipeline configuration URL.
+   */
+  private <T extends LoggerContext> LogPipelineSpecification<T> load(Provider<T> contextProvider,
+                                                                     URL configURL) throws JoranException {
+    // Create one AppenderContext per config file
+    T context = contextProvider.get();
+
+    CConfiguration pipelineCConf = CConfiguration.copy(cConf);
+    LogPipelineConfigurator configurator = new LogPipelineConfigurator(pipelineCConf);
+    configurator.setContext(context);
+
+    configurator.doConfigure(configURL);
+
+    // Check if the configuration has any error in it.
+    if (!new StatusChecker(context).isErrorFree(Status.ERROR)) {
+      List<Status> errors = new ArrayList<>();
+      for (Status status : context.getStatusManager().getCopyOfStatusList()) {
+        if (status.getEffectiveLevel() == Status.ERROR) {
+          errors.add(status);
+        }
+      }
+      throw new JoranException("Configuration failed " + errors);
+    }
+
+    // Default the pipeline name to the config file name if it is not set
+    if (context.getName() == null) {
+      String path = configURL.getPath();
+      int idx = path.lastIndexOf("/");
+      context.setName(idx < 0 ? path : path.substring(idx + 1));
+    }
+
+    byte[] checkpointPrefix = Bytes.toBytes(context.getName());
+    String prefixNum = configurator.getExecutionContext().getProperty(Constants.Logging.PIPELINE_CHECKPOINT_PREFIX_NUM);
+    if (prefixNum != null) {
+      try {
+        checkpointPrefix = Bytes.toBytes(Integer.parseInt(prefixNum));
```

Comment:
so if users are not using the `prefix_num` changing the context name will make them read the kafka messages from the beginning right ? we might want to document these properties properly if that is the case.